bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/internal/SecureHttpClientFactory.java

```diff
@@ -0,0 +1,278 @@
+            }
+        } else {
+            logger.warn("ignoring invalid type {} for parameter {}", value.getClass().getName(), parameter);
+            return defaultValue;
+        }
+
+    }
+
+    private synchronized void initialize() {
+        if (threadPool == null || commonHttpClient == null) {
+            try {
+                AccessController.doPrivileged(new PrivilegedExceptionAction<Void>() {
+                    @Override
+                    public Void run() {
+                        if (threadPool == null) {
+                            threadPool = createThreadPool("common", minThreadsShared, maxThreadsShared,
+                                    keepAliveTimeoutShared);
+                        }
+
+                        if (commonHttpClient == null) {
+                            commonHttpClient = createHttpClientInternal("common", null);
+                            commonHttpClientFacade = new HttpClientDelegate(commonHttpClient);
+                            logger.info("jetty shared http client created");
+                        }
+
+                        return null;
+                    }
+                });
+            } catch (PrivilegedActionException e) {
+                Throwable cause = e.getCause();
+                if (cause instanceof RuntimeException) {
+                    throw (RuntimeException) cause;
+                } else {
+                    throw new RuntimeException(cause);
+                }
+            }
+        }
+    }
+
+    private HttpClient createHttpClientInternal(String consumerName, String endpoint) {
+        try {
+            return AccessController.doPrivileged(new PrivilegedExceptionAction<HttpClient>() {
+                @Override
+                public HttpClient run() {
+
+                    logger.info("creating httpClient for endpoint {}", endpoint);
+                    SslContextFactory sslContextFactory = new SslContextFactory();
+                    sslContextFactory.setEndpointIdentificationAlgorithm("HTTPS");
+                    if (endpoint != null && trustmanagerProvider != null) {
+                        Stream<TrustManager> trustManagerStream = trustmanagerProvider.getTrustManagers(endpoint);
+                        TrustManager[] trustManagers = trustManagerStream.toArray(TrustManager[]::new);
+                        if (trustManagers.length > 0) {
+                            logger.info(
+                                    "using custom trustmanagers (certificate pinning) for httpClient for endpoint {}",
+                                    endpoint);
+                            try {
+                                SSLContext sslContext = SSLContext.getInstance("TLS");
+                                sslContext.init(null, trustManagers, null);
+                                sslContextFactory.setSslContext(sslContext);
+                            } catch (Exception ex) {
+                                throw new RuntimeException(
```

Comment:
This exception will only happen, if the java runtime does not support TLS, so in practise it will never be thrown. I see no need for a custom exception.

Reply:
Can you then maybe catch a specific exception? That would be much better, but I agree that the Jetty http-client makes that very hard by also ignoring these best practices.

Reply:
> This exception will only happen, if the java runtime does not support TLS

So, perhaps `throw new IllegalStateException("Your JRE needs to support TLS", ex);`?

Reply:
I was in doubt suggesting it 😄 

But the docblock did not make me happy enough, but I think its way better than a `RuntimeException`

> Signals that a method has been invoked at an illegal or inappropriate time. In other words, the Java environment or Java application is not in an appropriate state for the requested operation.